src/Evolution/Systems/ForceFree/System.hpp

```diff
@@ -0,0 +1,178 @@
+ *
+ * In SpECTRE, we evolve 'extended' (or augmented) version of Maxwell equations
+ * with two divergence cleaning scalar fields \f$\psi\f$ and \f$\phi\f$ :
+ *
+ * \f{align*}{
+ *  \nabla_a(F^{ab}+g^{ab}\psi) & = -J^b + \kappa_\psi n^b \psi \\
+ *  \nabla_a(^* F^{ab}+ g^{ab}\phi) & = \kappa_\phi n^b \phi
+ * \f}
+ *
+ * which reduce to the original Maxwell equations when \f$\psi=\phi=0\f$. For
+ * damping constants $\kappa_{\psi, \phi} > 0$, Gauss constraint violations are
+ * damped with timescales $\kappa_{\psi,\phi}^{-1}$ and propagated away.
+ *
+ * We decompose the EM field tensor as follows
+ * \f{align*}{
+ *  F^{ab} = n^a E^b - n^b E^a - \epsilon^{abcd}B_c n_d,
+ * \f}
+ *
+ * where $n^a$ is the normal to spatial hypersurface, $E^a$ and $B^a$ are
+ * electric and magnetic fields.
+ *
+ * Evolved variables are
+ *
+ * \f{align*}{
+ * \mathbf{U} = \sqrt{\gamma}\left[\,\begin{matrix}
+ *      E^i \\
+ *      B^i \\
+ *      \psi \\
+ *      \phi \\
+ *      q \\
+ * \end{matrix}\,\right] \equiv \left[\,\,\begin{matrix}
+ *      \tilde{E}^i \\
+ *      \tilde{B}^i \\
+ *      \tilde{\psi} \\
+ *      \tilde{\phi} \\
+ *      \tilde{q} \\
+ * \end{matrix}\,\,\right] \f}
+ *
+ * where \f$E^i\f$ is electric field, \f$B^i\f$ is magnetic field, $\psi$ is
+ * electric divergence cleaning field, $\phi$ is magnetic divergence cleaning
+ * field, \f$q\equiv-n_aJ^a\f$ is electric charge density, and \f$\gamma\f$ is
+ * the determinant of spatial metric.
+ *
+ * Corresponding fluxes \f$\mathbf{F}^j\f$ are
+ *
+ * \f{align*}{
+ *  F^j(\tilde{E}^i) & = -\beta^j\tilde{E}^i + \alpha
+ *      (\gamma^{ij}\tilde{\psi} - \epsilon^{ijk}_{(3)}\tilde{B}_k) \\
+ *  F^j(\tilde{B}^i) & = -\beta^j\tilde{B}^i + \alpha (\gamma^{ij}\tilde{\phi} +
+ *      \epsilon^{ijk}_{(3)}\tilde{E}_k) \\
+ *  F^j(\tilde{\psi}) & = -\beta^j \tilde{\psi} + \alpha \tilde{E}^j \\
+ *  F^j(\tilde{\phi}) & = -\beta^j \tilde{\phi} + \alpha \tilde{B}^j \\
+ *  F^j(\tilde{q}) & = \alpha \sqrt{\gamma}J^j - \beta^j \tilde{q}
+ * \f}
+ *
+ * and source terms are
+ *
+ * \f{align*}{
+ *  S(\tilde{E}^i) &= -\alpha \sqrt{\gamma} J^i - \tilde{E}^j \partial_j \beta^i
+ *    + \tilde{\psi} ( \gamma^{ij} \partial_j \alpha - \alpha \gamma^{jk}
+ *      \Gamma^i_{jk} ) \\
+ *  S(\tilde{B}^i) &= -\tilde{B}^j \partial_j \beta^i + \tilde{\phi} (
+ *      \gamma^{ij} \partial_j \alpha - \alpha \gamma^{jk} \Gamma^i_{jk} ) \\
+ *  S(\tilde{\psi}) &= \tilde{E}^k \partial_k \alpha + \alpha \tilde{q} -
+ *      \alpha \tilde{\phi} ( K + \kappa_\phi ) \\
+ *  S(\tilde{\phi}) &= \tilde{B}^k \partial_k \alpha - \alpha \tilde{\phi}
+ *      (K + \kappa_\phi ) \\
+ *  S(\tilde{q}) &= 0
+ * \f}
+ *
+ * See the documentation of Fluxes and Sources for further details.
+ *
+ * In addition to Maxwell equations, general relativistic force-free
+ * electrodynamics (GRFFE) assumes the following which are called the force-free
+ * (FF) conditions.
+ *
+ * \f{align*}{
+ *  F^{ab}J_b & = 0, \\
```

Comment:
Could you expand these to also be in terms of E & B?

Reply:
Sure

Reply:
E_i B^i =0

E_i E^i - B_i B^i <0.

Maybe state F.J=0 directly in terms of the drift current?